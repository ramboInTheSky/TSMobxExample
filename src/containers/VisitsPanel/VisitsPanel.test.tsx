import React from 'react'
import { shallow } from 'enzyme'
import { createShallow } from '@material-ui/core/test-utils'

import { VisitsPanel, VisitsPanelProps } from '.'

describe('<VisitsPanel />', () => {
	let props: VisitsPanelProps
	let shallowMaterial: any
	beforeEach(() => {
		props = {
			visits: [],
		}
		shallowMaterial = createShallow()
	})
	it('renders correctly with no data', () => {
		expect(shallow(<VisitsPanel {...props} />)).toMatchSnapshot()
	})

	it('renders correctly with data', () => {
		const newProps = {
			...props,
			visits: [
				{
					id: 'q3w4c5-w34cr-w34cw34-5cw3-4cw345-3cw34t',
					from: '2015-03-25T12:00:00Z',
					to: '2015-03-25T12:00:00Z',
					tags: ['123123', '234234', '345345'],
					lockGroups: ["Default", "Boiler", "1.11"],
				},
				{
					id: 'vvv4c5-w34cr-w34cw34-5cbb-4cw345-3cw311',
					from: '2019-04-01T10:00:10.000Z',
					to: '2019-05-01T10:00:10.000Z',
					tags: ['123123', '234234', '345345'],
					lockGroups: ["Default", "Boiler", "1.11"],
				},
			],
		}
		expect(shallow(<VisitsPanel {...newProps as any} />)).toMatchSnapshot()
	})
})
