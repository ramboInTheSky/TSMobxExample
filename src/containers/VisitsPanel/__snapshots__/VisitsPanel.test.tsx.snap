// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VisitsPanel /> renders correctly with data 1`] = `
<Boundary>
  <Styled(div)>
    <CSSTransitionGroup
      key="NaN"
      transitionAppear={true}
      transitionAppearTimeout={200}
      transitionEnter={true}
      transitionEnterTimeout={200}
      transitionLeave={true}
      transitionLeaveTimeout={200}
      transitionName="expand"
    >
      <Styled(WithStyles(Paper))
        style={
          Object {
            "opacity": 0.6,
          }
        }
      >
        <Styled(div)>
          <Styled(div)>
            <Styled(span)>
              this is a date - this is a date
            </Styled(span)>
            <Styled(span)>
              this is a time - this is a time
            </Styled(span)>
          </Styled(div)>
          <Styled(div)>
            Default, Boiler, 1.11
          </Styled(div)>
        </Styled(div)>
        <Styled(div)>
          Inactive
        </Styled(div)>
      </Styled(WithStyles(Paper))>
    </CSSTransitionGroup>
    <CSSTransitionGroup
      key="NaN"
      transitionAppear={true}
      transitionAppearTimeout={200}
      transitionEnter={true}
      transitionEnterTimeout={200}
      transitionLeave={true}
      transitionLeaveTimeout={200}
      transitionName="expand"
    >
      <Styled(WithStyles(Paper))
        style={
          Object {
            "opacity": 0.6,
          }
        }
      >
        <Styled(div)>
          <Styled(div)>
            <Styled(span)>
              this is a date - this is a date
            </Styled(span)>
            <Styled(span)>
              this is a time - this is a time
            </Styled(span)>
          </Styled(div)>
          <Styled(div)>
            Default, Boiler, 1.11
          </Styled(div)>
        </Styled(div)>
        <Styled(div)>
          Inactive
        </Styled(div)>
      </Styled(WithStyles(Paper))>
    </CSSTransitionGroup>
  </Styled(div)>
</Boundary>
`;

exports[`<VisitsPanel /> renders correctly with no data 1`] = `
<Boundary>
  <Styled(div) />
</Boundary>
`;
